modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTask.java

```diff
@@ -201,14 +198,11 @@ public MarketDataRequirements requirements() {
   /**
    * Returns an optional containing the first currency from the arguments or empty if both arguments are empty.
    */
-  private static Optional<Currency> reportingCurrency(Optional<Currency> ccy1, Optional<Currency> ccy2) {
-    if (ccy1.isPresent()) {
-      return ccy1;
-    }
-    if (ccy2.isPresent()) {
-      return ccy2;
+  private Optional<Currency> reportingCurrency() {
+    if (reportingCurrency.getCurrency().isPresent()) {
```

Comment:
No, the flatMap() function isn't invoked if the optional is empty. There is no pretty way to do this, I tried.